File: 153-FindMinimuminRotatedSortedArray/153-FindMinimuminRotatedSortedArray.cpp
// Last updated: 7/27/2026, 10:29:18 PM
1class Solution {
2public:
3    int findMin(vector<int>& nums) {
4        int n = nums.size();
5        int l = 0;
6        int r = n - 1;
7        
8        while (l < r) {
9            int mid = l + (r - l) / 2;  
10            
11            if (nums[mid] > nums[r]) {
12                l = mid + 1;  
13            } else {
14                r = mid;  
15            }
16        }
17        return nums[l];
18    }
19};
20

File: 153-FindMinimuminRotatedSortedArray/153-FindMinimuminRotatedSortedArray.java
// Last updated: 8/8/2026, 9:57:40 PM
1class Solution {
2    public int findMin(int[] nums) {
3        int l = 0;
4        int r = nums.length - 1;
5
6        while(l < r) {
7            int mid = l + (r - l) / 2;
8
9            if(nums[mid] > nums[r]) {
10                l = mid + 1;
11            } else {
12                r = mid;
13            }
14        }
15        return nums[r];
16    }
17}